src/shared/useAccessibilityAutoFocus.ts

```diff
@@ -24,34 +24,42 @@ export const focusOnElement = (elementRef: any) => {
 export const useAccessibilityAutoFocus = (isActive = true) => {
   const [autoFocusRef, setAutoFocusRef] = useState<any>();
 
-  const {callable, cancelable} = useMemo(() => {
-    return createCancellableCallbackPromise(
-      () => AccessibilityInfo.isScreenReaderEnabled().then(delay(AUTO_FOCUS_DELAY)),
-      isScreenReaderEnabled => {
-        if (!isScreenReaderEnabled) {
-          return;
-        }
-        focusOnElement(autoFocusRef);
-      },
-    );
-  }, [autoFocusRef]);
+  const {isScreenReaderEnabled} = useAccessibilityService();
+  const [isFocus, setIsFocus] = useState(false);
+  const [isLayoutUpdated, setIsLayoutUpdated] = useState(false);
+
+  useLayoutEffect(() => {
+    setIsLayoutUpdated(true);
+    return () => {
+      setIsLayoutUpdated(false);
+    };
+  }, []);
+
+  useFocusEffect(
+    useCallback(() => {
+      setIsFocus(true);
+      return () => {
+        setIsFocus(false);
+      };
+    }, []),
+  );
 
   useLayoutEffect(() => {
-    if (!autoFocusRef || !isActive) {
+    if (!isScreenReaderEnabled || !isActive || !isFocus || !isLayoutUpdated || !autoFocusRef) {
       return;
     }
 
-    callable();
-    return cancelable;
-  }, [callable, cancelable, autoFocusRef, isActive]);
+    // Call focus as soon as all considition is met
+    focusOnElement(autoFocusRef);
 
-  return [autoFocusRef, setAutoFocusRef];
-};
+    // Attempt to call it again just in case AccessibilityInfo.setAccessibilityFocus is delayed
+    const timeoutId = setTimeout(() => {
```

Comment:
I tried commenting this out and discovered that we still need this to make auto focus work on the 1st page of the how it works carousel on iOS.

Reply:
Right. Somehow `AccessibilityInfo.setAccessibilityFocus` is delayed. It needs timeout to actually focus on the element.